oneflow/core/framework/nn_graph.cpp

```diff
@@ -268,7 +268,16 @@ Maybe<void> MakeEagerBlobObjectList(std::vector<std::shared_ptr<vm::EagerBlobObj
   for (const auto& tensor : tensor_list) {
     CHECK_OR_RETURN(tensor->is_eager());
     if (tensor->is_consistent()) {
-      blob_list->push_back(JUST(JUST(tensor->cur_rank_phy_tensor())->eager_blob_object()));
+      const auto& parallel_id =
+          JUST(GetParallelId4CurrentProcessCtx(JUST(tensor->parallel_desc())));
+      if (parallel_id->has_value()) {
+        blob_list->push_back(JUST(JUST(tensor->cur_rank_phy_tensor())->eager_blob_object()));
+      } else {
+        // NOTE(chengcheng):
+        //   ConsistentTensor has NO phy tensor in current rank, so RunLazyJobInstrucntion
+        //   should filter this tensor and NOT send JobInstance for this pull/push callback.
+        blob_list->push_back(std::shared_ptr<vm::EagerBlobObject>());
```

Comment:
这里直接这样插入一个  empty （nullptr） 是有问题的，现象是直接 core dump，连 RunLazyJobInstructionType 的 Compute 都不会进去。


```console
chengcheng@oneflow-21:~/debug/graph $ ONEFLOW_TEST_DEVICE_NUM=2 python3 -m oneflow.distributed.launch --nproc_per_node 2 test_graph_buffer_limit.py 
Nothing happened because environment has been initialized
Nothing happened because environment has been initialized
tensor(...,
       placement=oneflow.placement(device_type="cuda", machine_device_ids={0 : [1]}, hierarchy=(1,)),
       sbp=(oneflow.sbp.broadcast,), is_lazy='True', size=(16, 10),
       dtype=oneflow.float32)
tensor(...,
       placement=oneflow.placement(device_type="cuda", machine_device_ids={0 : [1]}, hierarchy=(1,)),
       sbp=(oneflow.sbp.broadcast,), is_lazy='True', size=(16, 10),
       dtype=oneflow.float32)
E0911 11:16:38.016124 27739 nn_graph.cpp:310]  cclog: in rank 0 input has phy tensor. 
E0911 11:16:38.016373 27739 nn_graph.cpp:319]  cclog: in rank 0 output has NO phy tensor. 
E0911 11:16:38.016330 27740 nn_graph.cpp:312]  cclog: in rank 1 input has NO phy tensor. 
E0911 11:16:38.017067 27740 nn_graph.cpp:317]  cclog: in rank 1 output has phy tensor. 
Killing subprocess 27739
Killing subprocess 27740
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/chengcheng/oneflow/python/oneflow/distributed/launch.py", line 211, in <module>
    main()
  File "/home/chengcheng/oneflow/python/oneflow/distributed/launch.py", line 199, in main
    sigkill_handler(signal.SIGTERM, None)
  File "/home/chengcheng/oneflow/python/oneflow/distributed/launch.py", line 167, in sigkill_handler
    raise subprocess.CalledProcessError(
subprocess.CalledProcessError: Command '['/usr/bin/python3', '-u', 'test_graph_buffer_limit.py']' died with <Signals.SIGSEGV: 11>.
```